### PR TITLE
Add proper metrics for IDE startup (code browser)

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Add proper metrics for IDE startup (code browser)

See also [[internal chat]](https://gitpod.slack.com/archives/C01KGM9BH54/p1668757178295269?thread_ts=1668748433.738219&cid=C01KGM9BH54)

## Related Issue(s)

Fixes #

## How to test

- Start workspace in preview env
- Check if ide-metrics report `gitpod_ide_starthelper_phase_duration`

**How to check ide-metrics**
- Open this PR with Gitpod
```shell
# get ide metrics pod name
kubectl get pods | grep ide-metrics

# port forward
kubectl port-forward pods/<pod_name> 9500:9500
```
- Open 9500 in the browser

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/supervisor/pkg/metrics/metrics.go [hunk 1]
```diff
@@ -33,12 +33,14 @@ type GrpcMetricsReporter struct {
...
-			"grpc_server_msg_received_total": true,
-			"grpc_server_msg_sent_total":     true,
-			"grpc_server_started_total":      true,
-			"grpc_server_handling_seconds":   true,
+			"grpc_server_handled_total":             true,
+			"grpc_server_msg_received_total":        true,
+			"grpc_server_msg_sent_total":            true,
+			"grpc_server_started_total":             true,
+			"grpc_server_handling_seconds":          true,
+			"gitpod_ide_starthelper_phase_duration": true,
```
Comment (akosyakov) [inline components/supervisor/pkg/metrics/metrics.go:43]: I am not sure about the component and metric name. Everything is helper, maybe `ide-startup-timings`? `gitpod_ide_startup_duration` metric  with `phase` and `ide` as labels? something more meaningful than helper?

Comment (akosyakov) [reply components/supervisor/pkg/metrics/metrics.go:43]: Should we also measure it always btw or only on first run?

File: components/ide/starthelper/pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		addr = "localhost:22999"
+	}
+	pusher := push.New(addr, "codehelper")
+	phaseHistogram := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "gitpod_ide_starthelper_phase_duration",
+		Help:    "Duration of different phase during code startup",
+		Buckets: []float64{0.01, 0.1, 0.2, 0.5, 2, 5, 10, 30, 60},
+	}, []string{"phase", "ide"})
+
+	pusher.Collector(phaseHistogram)
```
Comment (akosyakov) [inline components/ide/starthelper/pkg/metrics/metrics.go:41]: is not it going to be expensive if we push it through supervisor again? We wanted to get rid of push gateway from supervisor and push directly to IDE proxy

**Comment by mustard-mh** [comment]
Close to go with light detection first https://github.com/gitpod-io/gitpod/pull/14790

**Pull request closed without merging**